### ISSUE TITLE
MismatchingDocblockParamType seen when union type is left out of `@param`

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#560]
Expected: no error

Observed: `MismatchingDocblockParamType - src/missing_phpdoc.php:8:8 - Parameter $x has wrong type 'mixed', should be 'array<mixed, mixed>'`

```php
<?php

class Foo {
    /**
     * Strangely, this doesn't warn if there's a description after $x in the below phpdoc tag
     *
     * @param int $i
     * @param $x
     */
    private static function example(int $i, array $x) : void {
        var_export($x);
        echo $i;
    }
}
```

**Comment by muglug** [comment]
Agreed - if the type isn't present, it should use whatever's in the signature

**Comment by TysonAndre** [comment]
A second example that's mostly solved if you just skip over an absent type: Psalm infers that the type is `mixed`, not `string`, inside the method body. So it doesn't emit a warning inside the method.

It'd be convenient to just ignore less specific or incorrect types for scalar types such as `?string`, `int`, etc. (Making exceptions for compatible types such as `@param callable`)

```php
<?php

/**
 * @param $str
 * @psalm-suppress MismatchingDocblockParamType
 */
function test(string $str) : void {
    echo count($str);
}
```
